x-pack/filebeat/tests/system/test_http_endpoint.py

```diff
@@ -308,6 +308,144 @@ def test_http_endpoint_invalid_hmac(self):
         assert r.status_code == 401
         self.assertRegex(r.json()['message'], 'invalid HMAC signature')
 
+    def test_http_endpoint_preserve_original_event(self):
+        """
+        Test http_endpoint input while preserving the original event.
+        """
+        options = """
+  preserve_original_event: true
+"""
+        self.get_config(options)
+        filebeat = self.start_beat()
+        self.wait_until(lambda: self.log_contains("Starting HTTP server on {}:{}".format(self.host, self.port)))
+
+        message = "somerandommessage"
+        payload = {self.prefix: message}
+        bp = json.dumps(payload, indent=2).encode('utf-8')
+        headers = {"Content-Type": "application/json"}
+        r = requests.post(self.url, headers=headers, data=json.dumps(payload))
+
+        filebeat.check_kill_and_wait()
+        output = self.read_output()
+
+        # To be able to do an exact comparison, it needs to be converted back into a dict, before reading it as bytes
```

Comment:
I think the value in this test is that it verifies that `event.original` matches the raw data that was sent (ignoring leading and trailing whitespace). So if the test unmarshals `event.original` and marshals it again then compares that to the original payload bytes then we haven't really varied that we preserved the source content.

What are the whitespace issues that required this? Can you trim the leading/trailing to remove possible newline issues and see if that works?